cpp/src/Slice/Parser.cpp

```diff
@@ -5780,27 +5763,31 @@ Slice::Operation::setExceptionList(const ExceptionList& el)
     }
 }
 
-Contained::ContainedType
-Slice::Operation::containedType() const
+ContainedList
+Slice::Operation::contents() const
 {
-    return ContainedTypeOperation;
+    ContainedList result;
+    result.insert(result.end(), _inParameters.begin(), _inParameters.end());
+    result.insert(result.end(), _outParameters.begin(), _outParameters.end());
+    return result;
+
 }
 
 bool
 Slice::Operation::uses(const ContainedPtr& contained) const
 {
     {
         ContainedPtr contained2 = ContainedPtr::dynamicCast(_returnType);
-        if(contained2 && contained2 == contained)
+        if (contained2 && contained2 == contained)
         {
             return true;
         }
     }
 
-    for (ExceptionList::const_iterator q = _throws.begin(); q != _throws.end(); ++q)
+    for (const auto& exception : _throws)
```

Comment:
Can we instead do a dynamic cast of contained to exception outside the loop, _throws can only contain exceptions and no point of iterating it if contained is not an exception

Reply:
Yes, and fixed.